### ISSUE TITLE
Support 64-bit integers in C on Windows

### ISSUE THREAD

**Issue by joshuaulrich** [joshuaulrich/xts#202]
The main issue with the lack of 64-bit integers at the C level on Windows is that `endpoints()` does not support sub-second resolution.  This is because the `long` C type is a 32-bit integer on Windows (even 64-bit Windows), but a 64-bit integer on widely-used unix-alikes.

```r
library(xts)
x <- structure(1:3, .Dim = c(3L, 1L), index = structure(c(1501072999.073, 
  1501072999.178, 1501073000.136), tzone = "", tclass = c("POSIXct", "POSIXt")),
  .indexCLASS = c("POSIXct", "POSIXt"), .indexTZ = "", tclass = c("POSIXct", "POSIXt"),
  tzone = "", class = c("xts", "zoo"))
endpoints(x, "ms", 500)  # [1] 0 3 on Windows, but should be 0, 2, 3
```

The solution is to add `#include <stdint.h>` to any file that needs a 64-bit integer, and then use the C99 type `int64_t`.  This may be an issue anywhere a cast to `(long)` is used.

**Comment by joshuaulrich** [comment]
Closing as fixed, since commits 26657859dc06d7f4f1984321aff7503bd4d8a95f and bb089480e2d00eeea2f06643fdc5d84603a9f81d seem to address the bulk of the issue.  There may still be some edge cases due to differences on other platforms.

**Comment by kboudt** [comment]
Thanks Joshua. This is a great improvement of the endpoints functionality, in particular for working with intraday price and quotes data with millisecond time stamp.
